synapse/rest/synapse/client/new_user_consent.py

```diff
@@ -61,6 +61,15 @@ async def _async_render_GET(self, request: Request) -> None:
             self._sso_handler.render_error(request, "bad_session", e.msg, code=e.code)
             return
 
+        # It should be impossible to get here without having first been through
+        # the pick-a-username step, which ensures chosen_localpart gets set.
+        if not session.chosen_localpart:
+            logger.warning("Session has no user name selected")
+            self._sso_handler.render_error(
+                request, "no_user", "No user name has been selected.", code=400
+            )
+            return
```

Comment:
My thinking for not asserting here is that I assume its possible for a client that is being stupid to get here, and so we don't want to get stack traces/sentry alerts?